src/makeInfinityAdder.js

```diff
@@ -4,7 +4,21 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  const innerFunction = (a) => {
+    const x = sum;
```

Comment:
The variable `x` is redundant, as you can directly return `sum` and then reset it when no arguments are passed to `innerFunction`. This will simplify your function's logic and improve its readability.